st2common/st2common/hooks.py

```diff
@@ -137,32 +153,83 @@ def _validate_token(request):
 
 class JSONErrorResponseHook(PecanHook):
     """
-    Hook which ensures that error response always contains JSON.
+    Handle all the errors and respond with JSON.
     """
 
-    def on_error(self, state, exc):
+    def on_error(self, state, e):
         request_path = state.request.path
         if cfg.CONF.api.serve_webui_files and request_path.startswith('/webui'):
             # We want to return regular error response for requests to /webui
             return
 
-        status_code = state.response.status
-        if status_code == httplib.NOT_FOUND:
-            message = 'The resource could not be found'
-        elif status_code == httplib.INTERNAL_SERVER_ERROR:
-            message = 'Internal Server Error'
+        LOG.debug('API call failed: %s' % (str(e)))
```

Comment:
Without a traceback turns out this log message isn't very useful. I will figure out a fix and create a PR